pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -392,6 +392,7 @@ def _check_device_config_and_set_final_flags(
         if self._devices_flag in ([], 0, "0"):
```

Comment:
why do we fall back here? shouldn't it be strict if someone's settings `0, [], '0'` explicitly?

Reply:
Nope! The logic hasn't been changed with the Accelerator Connector rewrite.

Also, a lot of users do `Trainer(accelerator="gpu", devices=torch.cuda.device_count())`, this should fall back to CPUAccelerator.

Reply:
why not fail? it's a classic example if misconfiguration no? I am trying to use something, but if I don't have it, framework should tell me.

Reply:
this would fail with the accelerator availability check. i agree with @rohitgr7 that we shouldn't fallback in this instance. in this case, "auto" is a better choice for users to select if they want whatever hardware is available

Reply:
Yes, we don't need this leniency now that `auto` is supported